tests/lib/rules/no-array-prototype-extensions.js

```diff
@@ -0,0 +1,260 @@
+'use strict';
+
+const rule = require('../../../lib/rules/no-array-prototype-extensions');
+const RuleTester = require('eslint').RuleTester;
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parser: require.resolve('@babel/eslint-parser'),
+  parserOptions: { ecmaVersion: 2020, sourceType: 'module' },
+});
+
+ruleTester.run('no-array-prototype-extensions', rule, {
+  valid: [
+    '[1, 2, 4].filter(el => { /* ... */ })',
+    'something.filter(el => { /* ... */ })',
+    "filterBy(something, 'foo')",
+    'this.filterBy()',
+    'something[0]',
+    'something[0].foo',
+    "get(something, 'foo.0.bar')",
+    "get(something, 'foo.0')",
+    "get(something, 'foo')[0]",
+    "get(something, 'firstObjects')",
+    "get(something, 'lastObjects')",
+    'something.length',
+    'something.firstObject()',
+    'something[something.length - 1]',
+    /** Replace is part of string native prototypes */
+    "'something'.replace()",
+    'something.replace()',
+    'something.isAny',
+    "something['compact']",
+  ],
+  invalid: [
```

Comment:
Can we ensure that we still detect these when optional chaining is used?

`arr?.firstObject?.foo`
`arr?.filterBy()`

Reply:
added, good point.